perfkitbenchmarker/benchmarks/cloud_bigtable_ycsb_benchmark.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs YCSB against Cloud Bigtable.
+
+Cloud Bigtable (https://cloud.google.com/bigtable/) is a managed NoSQL database
+with an HBase-compatible API.
+
+Compared to hbase_ycsb, this benchmark:
+  * Modifies hbase-site.xml to work with Cloud Bigtable.
+  * Adds the Bigtable client JAR.
+  * Adds alpn-boot-7.0.0.v20140317.jar to the bootclasspath, required to
+    operate.
+"""
+
+import os
+import posixpath
+
+from perfkitbenchmarker import data
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.packages import hbase
+from perfkitbenchmarker.packages import ycsb
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('google_bigtable_endpoint', 'bigtable.googleapis.com',
+                    'Google API endpoint for Cloud Bigtable.')
+flags.DEFINE_string('google_bigtable_admin_endpoint',
+                    'bigtabletableadmin.googleapis.com',
+                    'Google API endpoint for Cloud Bigtable table '
+                    'administration.')
+flags.DEFINE_string('google_bigtable_zone_name', 'us-central1-b',
+                    'Bigtable zone.')
+flags.DEFINE_string('google_bigtable_cluster_name', None,
+                    'Bigtable cluster name.')
+flags.DEFINE_string(
+    'google_bigtable_alpn_jar_url',
+    'http://central.maven.org/maven2/org/mortbay/jetty/alpn/'
+    'alpn-boot/7.0.0.v20140317/alpn-boot-7.0.0.v20140317.jar',
+    'URL for the ALPN boot JAR, required for HTTP2')
+flags.DEFINE_string(
+    'google_bigtable_hbase_jar_url',
+    'https://storage.googleapis.com/cloud-bigtable/jars/'
+    'bigtable-hbase/bigtable-hbase-0.1.5.jar',
+    'URL for the Bigtable-HBase client JAR.')
+
+
+BENCHMARK_INFO = {'name': 'cloud_bigtable_ycsb',
+                  'description': 'Run YCSB against Cloud Bigtable. '
+                  'configure the number of client VMs via --num_vms.',
+                  'scratch_disk': False,
+                  'num_machines': None}
```

Comment:
Wehn `num_machines` is `None`, it takes the value of `--num_vms`, defaulting to 1. This way, more client VMs can be used if desired.